Documentation/Design/Design Doc Template.md

```diff
@@ -0,0 +1,151 @@
+# Feature Name
+
+## Feature Overview
+
+A description of the feature.
```

Comment:
Maybe add examples of what would be included, such as functionality / reason for existence?